feature updates for tgocassisstitch

antonhibl opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail including motivation and any context -->

addresses #5125, this allows the user to optionally use either the prefix or
    suffix option when stitching frames together, just use OUTPUTSUFFIX instead
    of OUTPUTPREFIX to use it.
    
    closed #5132 in favor of opening this as to have a cleaner history and branch.

## Related Issue
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

#5125 

## How Has This Been Validated?
<!--- All changes need to be validated to confirm that they produce -->
<!--- scientifically accurate results. -->
<!--- If your changes include any new algorithms or changes to existing algorithms, -->
<!--- please indicate any publications or references for them. -->
<!--- If you manually validated the changes please indicate -->
<!--- what data you used and how you checked the results. -->

Tested locally and works by passing in either a OUTPUTSUFFIX or OUTPUT PREFIX.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- - [x] My code follows the code style of this project. -->
- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

antonhibl committed:
Squashing things

added suffix and prefix capabilities in tgocassisstitch

    addresses #5125, this allows the user to optionally use either the prefix or
    suffix option when stitching frames together, just use OUTPUTSUFFIX instead
    of OUTPUTPREFIX to use it.

minor fixes

typo

more typos

close to done, one error left

    need to figure out a replacement for ui.IsOptionSet as that is not actually
    a function. I need another way to test CLLI args.

final fixes to implement ui changes

fixed

    now employs a boolean flag to decide if the name parsed with out is a suffix
    or prefix, it defaults to a prefix.

adding some tests to confirm output runs as expected.

antonhibl committed:
adding to changelog

antonhibl committed:
modified to allow prefix, suffix, both, or neither for better extensibility

    it also maintains original value names so older scripts and pipelines will
    retain functionality.

adds another optional arg, cubename

    lets the user specify an optional cubename parameter which replaces the
    timestamp style naming for the output stitched cube. If the parameter is not
    given it defaults to the timestamp style naming convention for retaining
    existing functionality.

antonhibl committed:
added details to changelog

removed debugging code

added changelog notes to xml for tgocassisstitch

antonhibl committed:
added details to changelog

changelog fixes

removed typo

antonhibl committed:
typo in changelog

antonhibl committed:
fixes from review

added cubename argument logic

typo in else statement

antonhibl committed:
modifying tests

removed faulty/old tests based on makefiles

fixed single and multi frame tests for tgocassisstitch

remaining tests fixed, one on EFS to check with jenkins.

removed debug lines

updated to remove nil randomly appearing in tests

modified first stitch test to use fallback naming behaviour.

acpaquette commented:
@antonhibl Apologies for taking some time to get to this. Looks like it could use a fix for the changelog conflicts. The commit history here is also very messy. If we could do a `git rebase -i HEAD~33`, I think we could get it down to a handful of commits. I can help with that as well

acpaquette commented on isis/tests/FunctionalTestsTgocassisstitch.cpp:
```diff
@@ -37,7 +40,7 @@ TEST(TgoCassisstitch, TgoCassisstitchMultiframeTest) {
   }
 
-  Cube cube(prefix.path() + "/CAS-MCO-2016-11-22T16:16:16.833.cub");
+  Cube cube(prefix.path() + "/CAS-MCO-default.cub");
```
Could we only change the result of one of these tests? Basically only set the cubename in one of them rather than both?

antonhibl replied:
should be addressed with the latest commit

acpaquette replied:
Looks like this got unresolved?

antonhibl replied:
should be fixed now, only the multi frame test uses the default naming scheme

acpaquette commented on isis/src/tgo/apps/tgocassisstitch/tgocassisstitch.cpp:
```diff
@@ -66,20 +66,37 @@ namespace Isis {
+          FileName frameFileName(outputPrefBaseName +
+                                 frameIdentifier +
+                                 outputSuffBaseName + ".cub");
+          stitchFrame( frameMap.values(frameKey), frameFileName );
+          stitchProgress.CheckStatus();
```
Can we move this outside of the if block logic? You'll also need to moved the frameFileName declaration outside of the if block so it can be accessed outside of the logic block

antonhibl replied:
should be addressed with latest commit

acpaquette reviewed (commented):
Couple things to address outside of the re-base but overall looks good and tests are passing!

antonhibl committed:
simplified some logic and removed repeat lines in different logic branches

removing redundant lines from 8.0 release

addresses prefix and filename convention topical issues

acpaquette commented on isis/tests/FunctionalTestsTgocassisstitch.cpp:
```diff
@@ -22,12 +22,14 @@ TEST(TgoCassisstitch, TgoCassisstitchMultiframeTest) {
+
   QVector<QString> args = {"fromlist=" + cubeListFile,
-                           "outputprefix=" + prefix.path() + "/CAS-MCO"};
+                           "outputprefix=" + prefix.path() + "/CAS-MCO-"};
```
One last thing, it's concerning that we are having to add dashes in manually for tests. We shouldn't need to change in the input to get the same output.

It seems like we should be placing a dash between the prefix and the cube name in the `frameFileName` in the tgocassisstitch app

antonhibl replied:
Addressed with my latest PR and I've just checked that tests were passing for me locally @acpaquette

antonhibl committed:
resolving changelog merge conflicts (I think)

added details to changelog

small typo in link

acpaquette commented on isis/tests/TgoCassisModuleTests.cpp:
```diff
@@ -138,7 +138,7 @@ TEST_F(TgoCassisModuleKernels, TgoCassisStitchUnstitch) {
 
   QVector<QString> stitchArgs = {"fromlist=" + cubeListFile,
-                "outputprefix=" + prefix.path() + "/stitched"};
+                "outputprefix=" + prefix.path() + "/stitched-"};
```
The additional dashes are still an issue here. If we have to add to the output path here this is considered a breaking change. Rather than change the tests, we need to make the code conform to the tests

antonhibl replied:
addressed with latest commits

acpaquette reviewed (dismissed):
Looks like things are still kinda wonky

antonhibl committed:
removed some dashes in tests and replaced a mistaken default filename

antonhibl committed:
fixing the changleog resolves

antonhibl committed:
modified to allow prefix, suffix, both, or neither for better extensibility

    it also maintains original value names so older scripts and pipelines will
    retain functionality.

antonhibl committed:
small test changes

antonhibl commented:
@acpaquette latest commits should have addressed the issues with dashes and I have tested locally to ensure they pass

antonhibl reviewed (commented):
this should be ready for re-review @acpaquette

antonhibl committed:
cleaning up git artifacts left by squashing

acpaquette commented on CHANGELOG.md:
```diff
@@ -41,6 +41,11 @@ release.
+  outputsuffix, both, or neither for naming convention purposes. [#5162](https://github.com/DOI-USGS/ISIS3/pull/5162)
+- Updated download location for Dawn source files to include updated pck from HAMO Dawn mosaic [#4001](https://github.com/USGS-Astrogeology/ISIS3/issues/4001)
+- Pinned cspice version to 67 [#5083](https://github.com/USGS-Astrogeology/ISIS3/issues/5083)
+- Changed the `rsync` related commands in the ISIS SPICE Web Service document to `downloadIsisData` command
```
I think these need to be removed. I think they are changelog messages from other releases.

antonhibl committed:
changelog resolves

acpaquette reviewed (approved):
Sweet, looks good!
